src/field_theory/splitting_field.lean

```diff
@@ -109,46 +109,226 @@ namespace splitting_field_aux
+    mul_inv_cancel := have h : _ := @mul_inv_cancel, _,
+    exists_pair_ne := have h : _ := @exists_pair_ne, _,
+    nat_cast := splitting_field_aux.mk n ∘ (coe : ℕ → K),
+    nat_cast_zero := have h : _ := @comm_ring.nat_cast_zero, _,
+    nat_cast_succ := have h : _ := @comm_ring.nat_cast_succ, _,
+    nsmul := (•),
+    nsmul_zero' := have h : _ := @zero_nsmul, _,
+    nsmul_succ' := have h : _ := @succ_nsmul, _,
+    int_cast := splitting_field_aux.mk n ∘ (coe : ℤ → K),
+    int_cast_of_nat := have h : _ := @comm_ring.int_cast_of_nat, _,
+    int_cast_neg_succ_of_nat := have h : _ := @comm_ring.int_cast_neg_succ_of_nat, _,
+    zsmul := (•),
+    zsmul_zero' := have h : _ := @subtraction_comm_monoid.zsmul_zero', _,
+    zsmul_succ' := have h : _ := @subtraction_comm_monoid.zsmul_succ', _,
+    zsmul_neg' := have h : _ := @subtraction_comm_monoid.zsmul_neg', _,
+    rat_cast := splitting_field_aux.mk n ∘ (coe : ℚ → K),
+    rat_cast_mk := have h : _ := @field.rat_cast_mk, _,
+    qsmul := (•),
+    qsmul_eq_mul' := have h : _ := @field.qsmul_eq_mul', _,
+    npow := splitting_field_aux.npow n,
+    npow_zero' := have h : _ := @comm_ring.npow_zero', _,
+    npow_succ' := have h : _ := @comm_ring.npow_succ', _,
+    zpow := splitting_field_aux.zpow n,
+    zpow_zero' := have h : _ := @field.zpow_zero', _,
+    zpow_succ' := have h : _ := @field.zpow_succ', _,
+    zpow_neg' := have h : _ := @field.zpow_neg', _},
+  all_goals {
+    unfreezingI { induction n with n ih generalizing K },
+    { apply @h K },
+    { exact @ih _ _ _ } },
+end
+
+instance inhabited {n : ℕ} {f : K[X]} :
+  inhabited (splitting_field_aux n f) := ⟨37⟩
+
+/-- The injection from the base field as a ring homomorphism. -/
+@[simps] protected def mk_hom (n : ℕ) {K : Type u} [field K] {f : K[X]} :
+  K →+* splitting_field_aux n f :=
+{ to_fun := splitting_field_aux.mk n,
+  map_one' :=
+  begin
+    unfreezingI { induction n with k hk generalizing K },
+    { simp [splitting_field_aux.mk] },
+    exact hk
+  end,
+  map_mul' :=
+  begin
+    unfreezingI { induction n with k hk generalizing K },
+    { simp [splitting_field_aux.mk] },
+    intros x y,
+    change (splitting_field_aux.mk k) ((adjoin_root.of f.factor) _) = _,
+    rw [map_mul],
+    exact hk _ _
+  end,
+  map_zero' :=
+  begin
+    unfreezingI { induction n with k hk generalizing K },
+    { simp [splitting_field_aux.mk] },
+    change (splitting_field_aux.mk k) ((adjoin_root.of f.factor) 0) = _,
+    rw [map_zero, hk],
+  end,
+  map_add' :=
+  begin
+    unfreezingI { induction n with k hk generalizing K },
+    { simp [splitting_field_aux.mk] },
+    intros x y,
+    change (splitting_field_aux.mk k) ((adjoin_root.of f.factor) _) = _,
+    rw [map_add],
+    exact hk _ _
+  end }
+
+-- Note that all fields of this algebra instance are equal to the `{nat/int/rat}_cast` and
+-- `{n/z/q}smul`; this can be done by using `mk_hom`, definitionally equal to `mk`.
```

Comment:
I don't understand what you're trying to say about `mk_hom` here.

Reply:
This is what I mean; I think it's just a confusing message, so I removed it.